cmd/dev.go

```diff
@@ -399,13 +402,19 @@ func (cmd *DevCmd) startServices(f factory.Factory, config *latest.Config, gener
 	}
 
 	// Open UI if configured
-	if cmd.UI {
+	if cmd.UI || cmd.UIPort != 0 {
 		cmd.UI = false
 		logger.StartWait("Starting the ui server...")
 		defer logger.StopWait()
 
+		var port *int
+		if cmd.UIPort != 0 {
+			port = ptr.Int(cmd.UIPort)
```

Comment:
why not just use &cmd.UIPort?

Reply:
Because otherwise whenever cmd.UIPort == 0 the parameter forcePort isn't nil. It would be a pointer to 0, which forces the port 0 to be used.